R/attr-nodes.R

```diff
@@ -0,0 +1,83 @@
+# CURLY ------------------
+find_curly <- function(body, ns) {
+  i <- ".//md:text[not(@asis) and contains(text(), '{')]"
+  curlies <- xml2::xml_find_all(body, i, ns = ns)
+  # https://github.com/carpentries/pegboard/blob/a8db02ab037f2ffeab6e13cc3b662ea8c42822ad/R/get_images.R#L84
+  attr_texts <- xml2::xml_text(curlies)
+  no_closing <- !grepl("[}]", attr_texts)
+  if (any(no_closing)) {
+    close_xpath <- "self::*/following-sibling::md:text[contains(text(), '}')]"
+    for (not_closed in curlies[no_closing]) {
+      closing <- xml2::xml_find_all(
+        not_closed,
+        glue::glue("./{close_xpath}"),
+        ns
+      )
+      xml2::xml_text(not_closed) <- paste(
```

Comment:
should I add a test for the md -> xml + protect_curly() -> md round? If so is there an example?

Reply:
Yes, we should absolutely test this to make sure nothing on the xslt side suddenly changes. I think the easiest way to do this would be to use a file in a temporary directory and check the snapshot of the file:

https://github.com/ropensci/tinkr/blob/a577e9b4f3b94bcf09ead5c77ed49373a31db5d6/tests/testthat/test-class-yarn.R#L53-L63

Also, this is a clever solution 🤩! I had not thought about smashing together text into a text node separated by a newline like this! I always made them separate text nodes, but this solution solves so many problems.

Reply:
I was not sure whether it was a clever or lazy solution to be honest :joy: